### PR TITLE
[shuffle] Polish transfer NFT tutorial

### PR DESCRIPTION
**Pull request by sunmilee**
## Motivation

1) Added transfer e2e test tutorial for hello NFT
2) waitfortransfercompletion -> waitfortransfer 
3) Added nft type parameter to helper functions
4) Added move.ts access to console

### CODE REVIEW

File: shuffle/cli/src/console.rs [hunk 1]
```diff
@@ -26,7 +26,8 @@ pub fn handle(
         import * as helpers from "{helpers}";
         import * as main from "{main}";
         import * as codegen from "{codegen}";
-        import * as help from "{repl_help}";"#,
+        import * as help from "{repl_help}";
+        import * as move from "{mv}";"#,
```
Comment (dimroc) [inline shuffle/cli/src/console.rs:30]: nit: Everywhere else we import this as mv and do mv.Ascii. Let's change it for consistency? Or make everywhere move.ASCII again for consistency

File: shuffle/cli/tutorials/HelloNFT.md [hunk 2]
```diff
@@ -102,68 +102,197 @@ Let's try calling the `mint_nft` script function within the console.
...
+```
+import {
+  assert,
+  assertEquals,
+} from "https://deno.land/std@0.85.0/testing/asserts.ts";
+import * as devapi from "../main/devapi.ts";
+import * as main from "../main/mod.ts";
+import * as context from "../main/context.ts";
+import * as helpers from "../main/helpers.ts";
+import * as mv from "../main/move.ts";
```
Comment (dimroc) [inline shuffle/cli/tutorials/HelloNFT.md:189]: Like here. I do see in other spots we switch between move and mv.

File: shuffle/move/examples/e2e/message.test.ts [hunk 3]
```diff
@@ -59,15 +69,20 @@ Deno.test("Ability to set NFTs", async () => {
...
+  transferTxn = await devapi.waitForTransaction(transferTxn.hash);
   assert(transferTxn.success);
 
   // Check receiver has the nft
   const receiverNFTs = await main.decodedNFTs(secondUserContext.address);
   assertEquals(receiverNFTs[0].content_uri, contentUri);
   // Check sender nft_collection is empty
-  const senderNFTs = await main.decodedNFTs();
+  const senderNFTs = await main.decodedNFTs(
+    context.defaultUserContext.address,
```
Comment (dimroc) [inline shuffle/move/examples/e2e/message.test.ts:84]: Why are we passing in the default that it gets auto set to anyways?

File: shuffle/move/examples/e2e/message.test.ts [hunk 4]
```diff
@@ -27,25 +27,35 @@ Deno.test("Ability to set message", async () => {
...
   // Mint TestNFT into sender address
   const contentUri = "https://placekitten.com/200/300";
-
-  let txn = await main.createTestNFTScriptFunction(contentUri);
-  txn = await devapi.waitForTransactionCompletion(txn.hash);
+  let txn = await main.createTestNFTScriptFunction(
+    contentUri,
+    "TestNFT",
+    context.defaultUserContext,
+    context.defaultUserContext.address,
```
Comment (dimroc) [inline shuffle/move/examples/e2e/message.test.ts:56]: Same here. These last two lines are optional and with default are redundant to the defaults. Is for illustrative purposes?

File: shuffle/move/examples/e2e/message.test.ts [hunk 5]
```diff
@@ -27,25 +27,35 @@ Deno.test("Ability to set message", async () => {
...
+  );
+  senderInitializeTxn = await devapi.waitForTransaction(
     senderInitializeTxn.hash,
   );
   const secondUserContext = context.UserContext.fromEnv("test");
   let receiverInitializeTxn = await main.initializeNFTScriptFunction(
+    "TestNFT",
     secondUserContext,
+    context.defaultUserContext.address,
```
Comment (dimroc) [inline shuffle/move/examples/e2e/message.test.ts:42]: No need to send in the last parameter

File: shuffle/move/examples/e2e/message.test.ts [hunk 6]
```diff
@@ -27,25 +27,35 @@ Deno.test("Ability to set message", async () => {
 
 Deno.test("Ability to set NFTs", async () => {
   // Initialize nft_collection resource for both sender and receiver
-  let senderInitializeTxn = await main.initializeNFTScriptFunction();
-  senderInitializeTxn = await devapi.waitForTransactionCompletion(
+  let senderInitializeTxn = await main.initializeNFTScriptFunction(
+    "TestNFT",
+    context.defaultUserContext,
+    context.defaultUserContext.address,
```
Comment (dimroc) [inline shuffle/move/examples/e2e/message.test.ts:33]: Ditto

File: shuffle/cli/tutorials/HelloNFT.md [hunk 7]
```diff
@@ -102,68 +102,197 @@ Let's try calling the `mint_nft` script function within the console.
 
 > let typeArguments: string[] = [];
 
-> let args: any[] = [contentUri];
+> let args: any[] = [move.Ascii(contentUri)];
```
Comment (dimroc) [inline shuffle/cli/tutorials/HelloNFT.md:105]: I would prefer mv.MoveType (or move.MoveType), but then I realized you can remove the type annotation completely

**Review by dimroc — CHANGES_REQUESTED** [review]
Left some nits, let's do a pass on them and then land

File: shuffle/cli/tutorials/HelloNFT.md (hunk 7 above)
Comment (sunmilee) [reply shuffle/cli/tutorials/HelloNFT.md:105]: What do you mean remove the type annotation completely?

File: shuffle/move/examples/e2e/message.test.ts (hunk 3 above)
Comment (sunmilee) [reply shuffle/move/examples/e2e/message.test.ts:84]: Just so that the code is more explicit / for tutorial purposes

**Review by dimroc — APPROVED** [review]
/land

**Pull request merged**